website/facedetection/api/v2/views.py

```diff
@@ -0,0 +1,121 @@
+from django.conf import settings
+from django.db.models import Count, Q
+from django.utils import timezone
+
+from oauth2_provider.contrib.rest_framework import IsAuthenticatedOrTokenHasScope
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.generics import DestroyAPIView, ListAPIView, ListCreateAPIView
+from rest_framework.schemas.openapi import AutoSchema
+
+from photos.api.v2.serializers.photo import PhotoListSerializer
+from photos.models import Photo
+from thaliawebsite.api.v2.permissions import IsAuthenticatedOrTokenHasScopeForMethod
+from utils.media.services import fetch_thumbnails
+
+from .serializers import ReferenceFaceSerializer
+
+
+class YourPhotosView(ListAPIView):
+    serializer_class = PhotoListSerializer
+    permission_classes = [IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["photos:read", "facedetection:read"]
+
+    schema = AutoSchema(operation_id_base="FacedetectionMatches")
+
+    def get(self, request, *args, **kwargs):
+        if not request.member or request.member.current_membership is None:
+            raise PermissionDenied(
+                detail="You need to be a member in order to view your facedetection photos."
+            )
+        return self.list(request, *args, **kwargs)
+
+    def get_serializer(self, *args, **kwargs):
+        if len(args) > 0:
+            photos = args[0]
+            fetch_thumbnails([photo.file for photo in photos])
+        return super().get_serializer(*args, **kwargs)
+
+    def get_queryset(self):
```

Comment:
The original view has the same logic. We should probably deduplicate that so that changing this logic is not painful.